transactions/btc.ts

```diff
@@ -526,6 +527,201 @@ export async function getBtcFeesForNonOrdinalBtcSend(
+  changeAddress: string,
+  feeRate: number,
+): TransactionUtxoSelectionMetadata | undefined {
+  const dummyPrivateKey = '0000000000000000000000000000000000000000000000000000000000000001';
+
+  const inputSum = selectedUtxos.reduce<BigNumber>((sum, utxo) => sum.plus(utxo.value), new BigNumber(0));
+  const recipientTotal = recipients.reduce<BigNumber>(
+    (sum, recipient) => sum.plus(recipient.amountSats),
+    new BigNumber(0),
+  );
+
+  // these are conservative estimates
+  const ESTIMATED_VBYTES_PER_OUTPUT = 45; // actually around 50
+  const ESTIMATED_VBYTES_PER_INPUT = 85; // actually around 89 or 90
+  const BITCOIN_DUST_VALUE = 1000;
+
+  const estimatedFees = new BigNumber(feeRate).times(
+    ESTIMATED_VBYTES_PER_OUTPUT * recipients.length + ESTIMATED_VBYTES_PER_INPUT * selectedUtxos.length,
+  );
+  // ensure that the UTXOs can cover the expected outputs
+  if (!inputSum.gt(recipientTotal.plus(estimatedFees))) return undefined;
+
+  {
+    // try transaction with change
+    const tx = createTransaction(dummyPrivateKey, selectedUtxos, recipientTotal, recipients, changeAddress, 'Mainnet');
+
+    tx.sign(hex.decode(dummyPrivateKey));
+    tx.finalize();
+
+    const txSize = tx.vsize;
+    let sentSats = new BigNumber(0);
+
+    for (let i = 0; i < tx.outputsLength; i++) {
+      const output = tx.getOutput(i);
+      sentSats = sentSats.plus(new BigNumber((output.amount ?? 0n).toString()));
+    }
+
+    const change = sentSats.minus(recipientTotal);
+    const fee = new BigNumber(txSize).times(feeRate);
+
+    // check if there is change and if the change is greater than the vsize*feeRate + dust rate
+    if (change.gt(fee.plus(BITCOIN_DUST_VALUE))) {
+      const changeWithoutFee = change.minus(fee);
+      return {
+        selectedUtxos,
+        fee: fee.toNumber(),
+        feeRate: feeRate,
+        change: changeWithoutFee.toNumber(),
+      };
+    }
+  }
+
+  {
+    // try txn without change
+    const txNoChange = createTransaction(
+      dummyPrivateKey,
+      selectedUtxos,
+      recipientTotal,
+      recipients,
+      changeAddress,
+      'Mainnet',
+      true,
+    );
+
+    txNoChange.sign(hex.decode(dummyPrivateKey));
+    txNoChange.finalize();
+
+    const txSize = txNoChange.vsize;
+    let sentSats = new BigNumber(0);
+
+    for (let i = 0; i < txNoChange.outputsLength; i++) {
+      const output = txNoChange.getOutput(i);
+      sentSats = sentSats.plus(new BigNumber((output.amount ?? 0n).toString()));
+    }
+
+    const fee = inputSum.minus(sentSats);
+
+    if (fee.div(txSize).gt(feeRate)) {
+      return {
+        selectedUtxos,
+        fee: fee.toNumber(),
+        feeRate: fee.div(txSize).toNumber(),
+        change: 0,
+      };
+    }
+  }
+
+  return undefined;
+}
+
+function selectOptimalUtxos(
+  recipients: Recipient[],
+  selectedUtxos: UTXO[],
+  availableUtxos: UTXO[],
+  changeAddress: string,
+  feeRate: number,
+  currentBestUtxoCount: number | undefined = undefined,
```

Comment:
```suggestion
  currentBestUtxoCount?: number,
```
the default param only gets assigned if the param is undefined, so defaulting to undefined is a noop

Reply:
Oopsie 😅